File: app/src/main/java/com/dergoogler/liquidwars/server/NetInfo.java


package com.dergoogler.liquidwars.server;

import android.annotation.SuppressLint;
import android.net.wifi.WifiManager;
import android.content.Context;

public class NetInfo {
    public static String getIPAddress(Context context) {
        WifiManager wm = (WifiManager)context.getSystemService(Context.WIFI_SERVICE);
        int i = wm.getConnectionInfo().getIpAddress();
        return format(i);
    }

    public static String getNetMask(Context context) {
        WifiManager wm = (WifiManager)context.getSystemService(Context.WIFI_SERVICE);
        int i = wm.getDhcpInfo().netmask;
        return format(i);
    }

    public static String getBroadcastAddress(Context context) {
        WifiManager wm = (WifiManager)context.getSystemService(Context.WIFI_SERVICE);
        int ip = wm.getConnectionInfo().getIpAddress();
        int nm = wm.getDhcpInfo().netmask;
        return format(ip | (~nm));
    }

    public static String getSSID(Context context) {
        WifiManager wm = (WifiManager)context.getSystemService(Context.WIFI_SERVICE);
        return wm.getConnectionInfo().getSSID();
    }

    @SuppressLint("DefaultLocale")
    private static String format(int i) {
        return String.format("%d.%d.%d.%d",
            (i & 0xFF),
            ((i >> 8) & 0xFF),
            ((i >> 16) & 0xFF),
            ((i >> 24) & 0xFF));
    }
}

File: app/src/main/java/com/dergoogler/liquidwars/server/NetInfo.kt
package com.dergoogler.liquidwars.server

import android.annotation.SuppressLint
import android.content.Context
import android.net.wifi.WifiManager

object NetInfo {
    fun getIPAddress(context: Context): String {
        val wm = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val i = wm.connectionInfo.ipAddress
        return format(i)
    }

    fun getNetMask(context: Context): String {
        val wm = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val i = wm.dhcpInfo.netmask
        return format(i)
    }

    fun getBroadcastAddress(context: Context): String {
        val wm = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        val ip = wm.connectionInfo.ipAddress
        val nm = wm.dhcpInfo.netmask
        return format(ip or (nm.inv()))
    }

    fun getSSID(context: Context): String {
        val wm = context.getSystemService(Context.WIFI_SERVICE) as WifiManager
        return wm.connectionInfo.ssid
    }

    @SuppressLint("DefaultLocale")
    private fun format(i: Int): String {
        return String.format(
            "%d.%d.%d.%d",
            (i and 0xFF),
            ((i shr 8) and 0xFF),
            ((i shr 16) and 0xFF),
            ((i shr 24) and 0xFF)
        )
    }
}
